tests/testproject/TestProject.IntegrationServiceA/MongoDB/MongoDBExtensions.cs

```diff
@@ -0,0 +1,39 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using MongoDB.Driver;
+
+public static class MongoDBExtensions
+{
+    public static void MapMongoMovieApi(this WebApplication app)
+    {
+        app.MapGet("/mongodb/databases", GetDatabaseNamesAsync);
+
+        app.MapGet("/mongodb/movies", GetMoviesAsync);
+    }
+
+    private static async Task<List<string>> GetDatabaseNamesAsync(IMongoClient client, IMongoDatabase db)
```

Comment:
These handlers should probably take the abstract version of the resource as a dependency as a well via keyed DI. That way you can test both flavors.